libbeat/cfgfile/reload.go

```diff
@@ -63,13 +66,60 @@ func NewReloader(cfg *common.Config) *Reloader {
 	config := DefaultDynamicConfig
 	cfg.Unpack(&config)
 
+	path := config.Path
+	if !filepath.IsAbs(path) {
+		path = paths.Resolve(paths.Config, path)
+	}
+
 	return &Reloader{
 		registry: NewRegistry(),
 		config:   config,
+		path:     path,
 		done:     make(chan struct{}),
 	}
 }
 
+// Check configs are valid (only if reload is disabled)
+func (rl *Reloader) Check(runnerFactory RunnerFactory) error {
```

Comment:
It is quite similar to the existing config reload code in `Run` function. I would extract anything that could be extracted into smaller functions. So those could be reused it in both `Check` and `Run`. Thus, code duplication could be eliminated.

Reply:
I'll do some refactoring, thanks!